model/logs/gen-ai-events.yaml

```diff
@@ -0,0 +1,42 @@
+groups:
+  - id: gen_ai.common.event.attributes
+    type: attribute_group
+    brief: >
+      Describes common Gen AI event attributes.
+    attributes:
+      - ref: gen_ai.system
+
+  - id: gen_ai.system.message
+    name: gen_ai.system.message
+    type: event
+    brief: >
+      This event describes the instructions passed to the GenAI system inside the prompt.
+    extends: gen_ai.common.event.attributes
+
+  - id: gen_ai.user.message
+    name: gen_ai.user.message
+    type: event
+    brief: >
+      This event describes the prompt message specified by the user.
+    extends: gen_ai.common.event.attributes
+
+  - id: gen_ai.assistant.message
+    name: gen_ai.assistant.message
+    type: event
+    brief: >
+      This event describes the assistant message passed to GenAI system or received from it.
+    extends: gen_ai.common.event.attributes
+
+  - id: gen_ai.tool.message
+    name: gen_ai.tool.message
+    type: event
+    brief: >
+      This event describes the tool or function response message.
```

Comment:
> In [anthropic docs](https://docs.anthropic.com/en/docs/build-with-claude/tool-use#handling-tool-use-and-tool-result-content-blocks), they say this:

> Unlike APIs that separate tool use or use special roles like tool or function, Anthropic’s API integrates tools directly into the user and assistant message structure.

> In our spec, we hint there is a tool role (e.g. gen_ai.tool.message), but that's not the case in anthropic. Is that ok? What does cohere do?

As I mentioned in the other comment, we'll need to add a document that shows how to apply generic semconv to individual systems.

Given that OpenAI API is de-facto embraced by the community and differnt LLM vendors as universal one, we, based on the SIG discussions, decided to keep things more open-ai friendly.

For Anthropic it probably means that the response

```json
{
  "role": "assistant",
  "content": [
    {
      "type": "text",
      "text": "<thinking>To answer this question, I will: 1. Use the get_weather tool to get the current weather in San Francisco. 2. Use the get_time tool to get the current time in the America/Los_Angeles timezone, which covers San Francisco, CA.</thinking>"
    },
    {
      "type": "tool_use",
      "id": "toolu_01A09q90qw90lq917835lq9",
      "name": "get_weather",
      "input": {"location": "San Francisco, CA"}
    }
  ]
}
```

would be represented in choice event in the same way as openai one does:

```
`{"index":0,"finish_reason":"tool_calls","message":{"tool_calls":[{"id":"toolu_01A09q90qw90lq917835lq9","function":{"name":"get_weather","arguments":"{\"location\":\"San Francisco, CA\"}"},"type":"tool_use"}]}`
```